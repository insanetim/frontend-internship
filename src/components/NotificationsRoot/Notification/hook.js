import { useCallback, useEffect } from 'react'
import { notification } from 'antd'

const useContainer = props => {
  const { id, messageType, messageText, duration, hideNotification } = props

  const closeNotification = useCallback(() => {
    hideNotification(id)
    notification.destroy(id)
  }, [])

  useEffect(() => {
    notification[messageType]({
      key: id,
      message: messageText,
      duration,
      onClose: closeNotification
    })
  }, [])

  return { closeNotification }
}

export default useContainer
