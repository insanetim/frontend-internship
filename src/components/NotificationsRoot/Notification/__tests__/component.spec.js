import { shallow } from 'enzyme'

import Notification from '../component'

const mockedHookData = {}
jest.mock('../hook', () => jest.fn(() => mockedHookData))

describe('Notification component tests', () => {
  const props = {
    id: 'test/id',
    messageType: 'success',
    messageText: 'test/message',
    duration: 2.5,
    hideNotification: jest.fn()
  }

  const component = shallow(<Notification {...props} />)

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
