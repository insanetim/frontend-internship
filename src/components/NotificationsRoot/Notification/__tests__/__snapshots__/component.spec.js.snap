// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component tests matches snapshot 1`] = `""`;
